godel_process_planning/src/common_utils.h

```diff
@@ -0,0 +1,126 @@
+   * @return Tool pose corresponding to this surface point
+   */
+  Eigen::Affine3d createNominalTransform(const geometry_msgs::Pose& ref_pose,
+                                         const geometry_msgs::Point& pt);
+
+  /**
+   * @brief Given a path and robot model, this method creates a descartes planner and attempts to solve the path
+   * @param in_path Trajectory to solve
+   * @param robot_model Robot model used for IK/FK by planner
+   * @param out_path The solution path, if found, otherwise undefined
+   * @return True if path was found and result placed inside 'out_path'; False otherwise.
+   */
+  bool descartesSolve(const DescartesTraj& in_path,
+                      descartes_core::RobotModelConstPtr robot_model,
+                      DescartesTraj& out_path);
+  /**
+   * @brief Extracts joint position values from Descartes trajectory and packs them into a ROS message
+   * @param solution The Descartes trajectory used to generate nominal joint trajectory
+   * @param model The robot model used in generating the above trajectory
+   * @return A ROS joint trajectory with only the 'points' field filled in; you must add header/joint name info
+   */
+  trajectory_msgs::JointTrajectory toROSTrajectory(const DescartesTraj& solution,
+                                                   const descartes_core::RobotModel& model);
+  /**
+   * @brief Updates the joint names, frame id, and time stamp of the given trajectory
+   * @param joints Joint names; listed in same order as the values they correspond to
+   * @param traj The trajectory to modify
+   */
+  void fillTrajectoryHeaders(const std::vector<std::string>& joints,
+                             trajectory_msgs::JointTrajectory& traj);
+
+  /**
+   * @brief Attempts to capture the most recent JointState from the given topic
+   * @param topic The joint topic name on which to listen for the robot state
+   * @return The joint positions captured from the topic or a std::runtime_error
+   */
+  std::vector<double> getCurrentJointState(const std::string& topic);
+
+  /**
+   * @brief Creates descartes trajectory consisting of cartesian positions in a linear path between start and stop
+   * @param start The start pose of the linear interpolation
+   * @param stop The final pose of the linear interpolation
+   * @param ds The distance (m) between points in the linear path; defaults to 10cm
+   * @return A linear interpolated path from start to stop in Descartes format
+   */
+  DescartesTraj createLinearPath(const Eigen::Affine3d& start,
+                                 const Eigen::Affine3d& stop,
+                                 double ds = 0.1);
+  /**
+   * @brief Creates descartes trajectory consisting of joint interpolated positions from start to stop
+   * @param start The initial joint configuration
+   * @param stop The final joint configuration
+   * @param dtheta The distance (radians) between points in the interpolation; defaults to 1 degree
```

Comment:
I suppose it should read the `maximum` distance between points in the interpolation.